Update get_kinetic_matrix() with forward/backward difference formulas

Chriscrosser3310 opened the pull request:
I changed the way to generate Pentadiagonal matrix and applied this matrix with closed boundary conditions:
![image](https://user-images.githubusercontent.com/31495624/79955397-a1255b80-8433-11ea-959f-6f5e05a2b9bb.png)

Chriscrosser3310 committed:
Update single_electron.py

Chriscrosser3310 committed:
Apply forward/backward

Chriscrosser3310 committed:
Update comments

pedersor commented on single_electron.py:
```diff
@@ -217,39 +217,34 @@ def get_kinetic_matrix(self):
+            #replace two ends of the matrix with forward/backword formulas
+            mat[0, :6] = A_end_0
+            mat[-1, -6:] = A_end_0[::-1]
+            mat[0, :6] = A_end_1
```
problem when using 3-point method

pedersor commented on single_electron.py:
```diff
@@ -217,39 +217,34 @@ def get_kinetic_matrix(self):
         elif self.n_point_stencil == 3:
             A_central = [-2., 1.]
-            A_end = [2., -5., 4., -1.]
+            A_end_0 = [2., -5., 4., -1.]
```
I am not sure if this is the correct end-point formula for 3-point method

pedersor reviewed (commented):
I agree that this will develop the right matrix for the 5-point method. However, there are some small issues for the 3-point method. Please see code comments.

Chriscrosser3310 committed:
Merge remote-tracking branch 'upstream/master'

Chriscrosser3310 committed:
convergence test

1. add FDM algorithm
2. add script to get exact gs energy with grid number = 5000
3. modify convergence test

Chriscrosser3310 committed:
single_electron change EigenSolver matrix ends to 0

Changed 5-point and 3-point method matrix ends to 0 in the class EigenSolver

Chriscrosser3310 committed:
convergence test updated to be more user friendly

Changed the structure of convergence test so it's more user friendly.

New features:
- Can save images in the directory /convergence_test
- Made the main test as a function, which takes a Solver class, range, potential_fn,
boundary_condition and n_point_stencil as inputs.

Also made it show correct convergence rates for exp_hydrogenic and poschl_teller with 3-point and 5-point method, within range (-5, 5) and (-20, 20)
